### DISCUSSION TITLE
What happens if depends_on_java does not match?

### DISCUSSION THREAD

**Discussion by devton-here** [Homebrew/discussions#3145] [Writing Formulae/Casks] [answered]
I want to provide my app (a CLI) as a binary via homebrew and it depends on openjdk 8 (it won't work yet with another version)

1. What if a user installs my app but has already openjdk 11 or openjdk16 installed?

2. Can homebrew install openjdk8 automatically and make it available only for my app as the default java version?

3. My app comes a binary (jar) and there source code is not open source, can I still use a normal formulae or do I need make a cask?

**Comment by SMillerDev** [comment]
Is it open source?

**Reply by devton-here** [reply]
> Is it open source?

No, it's not.

**Comment by carlocab** [comment, marked as answer]
`depends_on_java` only works for casks. For formulae you can do `depends_on "openjdk@8"` which is the version you need. You can also configure your formula so that it uses the `openjdk@8` dependency and not any other JDK.

Since you provide a jar this should work fine for a formula. It will need to be hosted in an external tap, however, since we accept only open source software in homebrew-core.

**Reply by devton-here** [reply]
Okay, thanks for the clarification.

One more question: if I host it myself, how is it integrated then with homebrew. Or in other words: how does a user can install software when a tap is hosted externally. Does the user need to add my external repository manually?

**Reply by SMillerDev** [reply]
https://docs.brew.sh/Taps and https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap can answer that for you
